finesse/gui/opus_view.py

```diff
@@ -0,0 +1,226 @@
+        layout.addWidget(log_box)
+        return layout
+
+    def _create_status_page(self) -> QVBoxLayout:
+        """Creates the status_page.
+
+        Returns:
+            QHBoxLayout: The layout with the web view.
+        """
+        status_page = QGroupBox("Status")
+        self.status = QWebEngineView()
+        self.status.setMinimumSize(QSize(200, 200))
+
+        _layout = QVBoxLayout()
+        _layout.addWidget(self.status)
+        status_page.setLayout(_layout)
+
+        layout = QVBoxLayout()
+        layout.addWidget(status_page)
+        return layout
+
+    def url(self, action: str) -> str:
+        """Builds an URL out of the ip and the action.
+
+        Args:
+            action: The action to use.
+
+        Returns:
+            str: The constructed URL
+        """
+        return f"http://{self.ip}/{self.commands[action]}"
+
+    def on_acction_button_clicked(self, action: str) -> None:
+        """Execute the given action by sending a message to the appropriate topic.
+
+        TODO: Here assuming we are going to use pubsub or equivalent to send messages
+        around and communicate between backend and front end.
+
+        Args:
+            action: Action to be executed.
+        """
+        logging.info(f"OPUS action '{self.url(action)}' executed!")
+
+    def display_status(self) -> None:
+        """Retrieves and displays the new status.
+
+        TODO: I'm not sure who should populate the error log in the GUI. Probably the
+        ones handling the individual actions above. These are all placeholders, for now.
+        """
+        logging.info("Getting OPUS status!")
+        self.status.load(QUrl(self.url("status")))
+        self.status.show()
+
+        logging.getLogger("OPUS").error("Oh, no! Something bad happened!")
+
+    def open_opus(self) -> None:
+        """Opens OPUS front end somewhere else.
+
+        TODO: No idea what this is supposed to do.
+        """
+        logging.info("Going to OPUS!")
+
+
+class OPUSLogHandler(logging.Handler):
+    """Specific logger for the errors related to OPUS.
+
+    Only log messages using the OPUS logger will be recorded here. Typically, they will
+    be error messages, but it can be any information worth to be logged.
+    """
+
+    @classmethod
+    def set_handler(cls, log_area: QTextBrowser):
+        """Creates the handler and adds it to the logger.
+
+        Args:
+            log_area: The area where the log will be printed.
+        """
+        ch = cls(weakref.ref(log_area))
+
+        formatter = logging.Formatter(
+            fmt="%(asctime)s [%(levelname)s]: %(message)s", datefmt="%Y/%m/%d %H:%M:%S"
+        )
+        ch.setFormatter(formatter)
+
+        logging.getLogger("OPUS").addHandler(ch)
+
+    def __init__(self, log_area: weakref.ref):
+        """Constructor of the Handler.
+
+        Args:
+            log_area: A weak reference to the log area.
+        """
+        super(OPUSLogHandler, self).__init__()
```

Comment:
With Python 3 you can just do:

```suggestion
        super().__init__()
```